File: apple/Sources/Shared/Theme.swift
import SwiftUI

/// Shared palette — a calm dark theme matching the web client's tone.
/// Colors are platform-neutral SwiftUI `Color`s (no UIColor/NSColor) so the
/// same code compiles on iOS and macOS.
enum Theme {
    static let bg = Color(red: 0.07, green: 0.07, blue: 0.10)
    static let card = Color.white.opacity(0.04)
    static let cardBorder = Color.white.opacity(0.08)
    static let textPrimary = Color(red: 0.90, green: 0.92, blue: 0.95)
    static let textSecondary = Color(red: 0.63, green: 0.69, blue: 0.75)
    static let textMuted = Color(red: 0.42, green: 0.42, blue: 0.48)

    static let blue = Color(red: 0.39, green: 0.70, blue: 0.93)   // admissibility
    static let orange = Color(red: 0.96, green: 0.68, blue: 0.33) // competitiveness
    static let green = Color(red: 0.41, green: 0.83, blue: 0.57)  // fit / strong
    static let purple = Color(red: 0.62, green: 0.48, blue: 0.92) // confidence
    static let red = Color(red: 0.96, green: 0.41, blue: 0.41)    // high reach / flags

    /// Map a positioning label to its band color.
    static func bandColor(_ label: String?) -> Color {
        let l = (label ?? "").lowercased()
        if l.contains("highly competitive") { return green }
        if l.contains("high reach") { return red }
        if l.contains("reach") { return orange }
        if l.contains("competitive") { return blue }
        return textMuted
    }

    /// Map an evidence-confidence level to a color.
    static func confidenceColor(_ level: String?) -> Color {
        switch (level ?? "").lowercased() {
        case "high": return green
        case "medium": return orange
        default: return red
        }
    }
}

extension View {
    /// Standard card chrome used across the app.
    func ccCard() -> some View {
        self
            .padding(14)
            .background(Theme.card)
            .overlay(RoundedRectangle(cornerRadius: 10).stroke(Theme.cardBorder, lineWidth: 1))
            .clipShape(RoundedRectangle(cornerRadius: 10))
    }
}

File: windows/CollegeCounselor/Theme.cs
using System.Windows;
using System.Windows.Controls;
using System.Windows.Media;

namespace CollegeCounselor;

/// Shared dark palette + small builders so the code-built views stay terse and
/// consistent. Mirrors the SwiftUI app's Theme.
internal static class Theme
{
    public static readonly Brush Bg = Frozen(0x12, 0x12, 0x19);
    public static readonly Brush Card = Frozen(0x1B, 0x1B, 0x24);
    public static readonly Brush CardBorder = Frozen(0x2A, 0x2A, 0x38);
    public static readonly Brush TextPrimary = Frozen(0xE5, 0xEA, 0xF0);
    public static readonly Brush TextSecondary = Frozen(0xA0, 0xAE, 0xC0);
    public static readonly Brush TextMuted = Frozen(0x6B, 0x6B, 0x7A);

    public static readonly Brush Blue = Frozen(0x63, 0xB3, 0xED);   // admissibility
    public static readonly Brush Orange = Frozen(0xF6, 0xAD, 0x55); // competitiveness
    public static readonly Brush Green = Frozen(0x68, 0xD3, 0x91);  // fit / strong
    public static readonly Brush Purple = Frozen(0x9F, 0x7A, 0xEA); // confidence
    public static readonly Brush Red = Frozen(0xF5, 0x65, 0x65);    // reach / flags

    private static SolidColorBrush Frozen(byte r, byte g, byte b)
    {
        var br = new SolidColorBrush(Color.FromRgb(r, g, b));
        br.Freeze();
        return br;
    }

    public static Brush BandColor(string? label)
    {
        var l = (label ?? string.Empty).ToLowerInvariant();
        if (l.Contains("highly competitive")) return Green;
        if (l.Contains("high reach")) return Red;
        if (l.Contains("reach")) return Orange;
        if (l.Contains("competitive")) return Blue;
        return TextMuted;
    }

    public static Brush ConfidenceColor(string? level) => (level ?? "").ToLowerInvariant() switch
    {
        "high" => Green,
        "medium" => Orange,
        _ => Red,
    };

    // ─── Builders ────────────────────────────────────────────────────────
    public static TextBlock Label(string text, double size = 12, Brush? color = null, bool bold = false) => new()
    {
        Text = text,
        FontSize = size,
        Foreground = color ?? TextSecondary,
        FontWeight = bold ? FontWeights.SemiBold : FontWeights.Normal,
        TextWrapping = TextWrapping.Wrap,
        Margin = new Thickness(0, 0, 0, 4),
    };

    public static TextBox Input(string? text = null) => new()
    {
        Text = text ?? "",
        Background = Frozen(0x0F, 0x0F, 0x16),
        Foreground = TextPrimary,
        BorderBrush = CardBorder,
        CaretBrush = TextPrimary,
        Padding = new Thickness(8, 6, 8, 6),
        Margin = new Thickness(0, 0, 0, 10),
        FontSize = 13,
    };

    public static Button Button(string text, Brush? bg = null)
    {
        var b = new Button
        {
            Content = text,
            Padding = new Thickness(14, 8, 14, 8),
            Margin = new Thickness(0, 2, 8, 2),
            Foreground = bg == null ? TextPrimary : Frozen(0x0A, 0x0E, 0x17),
            Background = bg ?? Card,
            BorderBrush = CardBorder,
            FontSize = 13,
            Cursor = System.Windows.Input.Cursors.Hand,
        };
        return b;
    }

    public static Border Cardize(UIElement child) => new()
    {
        Background = Card,
        BorderBrush = CardBorder,
        BorderThickness = new Thickness(1),
        CornerRadius = new CornerRadius(10),
        Padding = new Thickness(14),
        Margin = new Thickness(0, 0, 0, 14),
        Child = child,
    };
}
